tests/queries/0_stateless/02317_distinct_in_order_optimization_explain.reference

```diff
@@ -50,70 +50,65 @@ algorithm: InOrder
-Sorting (Stream): a ASC, b ASC
-Sorting (Stream): a ASC, b ASC
+Sorting: a ASC, b ASC
+Sorting: a ASC, b ASC
 -- check that reading in order optimization for ORDER BY and DISTINCT applied correctly in the same query
 -- disabled, check that sorting description for ReadFromMergeTree match ORDER BY columns
-Sorting (Stream): a ASC
-Sorting (Stream): a ASC
-Sorting (Stream): a ASC
+Sorting: a ASC
+Sorting: a ASC
 -- enabled, check that ReadFromMergeTree sorting description is overwritten by DISTINCT optimization i.e. it contains columns from DISTINCT clause
-Sorting (Stream): a ASC, b ASC
-Sorting (Stream): a ASC, b ASC
-Sorting (Stream): a ASC, b ASC
+Sorting: a ASC
+Sorting: a ASC
+Sorting: a ASC, b ASC
+Sorting: a ASC, b ASC
 -- enabled, check that ReadFromMergeTree sorting description is overwritten by DISTINCT optimization, but direction used from ORDER BY clause
-Sorting (Stream): a DESC, b DESC
-Sorting (Stream): a DESC, b DESC
-Sorting (Stream): a DESC, b DESC
+Sorting: a DESC
+Sorting: a DESC
+Sorting: a DESC, b DESC
+Sorting: a DESC, b DESC
 -- enabled, check that ReadFromMergeTree sorting description is NOT overwritten by DISTINCT optimization (1), - it contains columns from ORDER BY clause
-Sorting (Stream): a ASC, b ASC
-Sorting (Stream): a ASC, b ASC
-Sorting (Stream): a ASC, b ASC
+Sorting: a ASC
+Sorting: a ASC, b ASC
+Sorting: a ASC
+Sorting: a ASC, b ASC
 -- enabled, check that ReadFromMergeTree sorting description is NOT overwritten by DISTINCT optimization (2), - direction used from ORDER BY clause
-Sorting (Stream): a DESC, b DESC
-Sorting (Stream): a DESC, b DESC
-Sorting (Stream): a DESC, b DESC
+Sorting: a DESC, b DESC
+Sorting: a DESC, b DESC
+Sorting: a DESC, b DESC
+Sorting: a DESC, b DESC
 -- enabled, check that disabling other 'read in order' optimizations do not disable distinct in order optimization
-Sorting (Stream): a ASC, b ASC
-Sorting (Stream): a ASC, b ASC
-Sorting (Stream): a ASC, b ASC
+Sorting: a ASC, b ASC
+Sorting: a ASC, b ASC
 === enable new analyzer ===
 -- enabled, check that sorting properties are propagated from ReadFromMergeTree till preliminary distinct
-Sorting (Stream): __table1.a ASC, __table1.b ASC
-Sorting (Stream): __table1.a ASC, __table1.b ASC
-Sorting (Stream): __table1.a ASC, __table1.b ASC
-Sorting (Stream): a ASC, b ASC
+Sorting: __table1.a ASC, __table1.b ASC
+Sorting: a ASC, b ASC
 -- disabled, check that sorting description for ReadFromMergeTree match ORDER BY columns
-Sorting (Stream): __table1.a ASC
-Sorting (Stream): __table1.a ASC
-Sorting (Stream): __table1.a ASC
-Sorting (Stream): a ASC
+Sorting: __table1.a ASC
+Sorting: a ASC
 -- enabled, check that ReadFromMergeTree sorting description is overwritten by DISTINCT optimization i.e. it contains columns from DISTINCT clause
-Sorting (Stream): __table1.a ASC, __table1.b ASC
-Sorting (Stream): __table1.a ASC, __table1.b ASC
-Sorting (Stream): __table1.a ASC, __table1.b ASC
-Sorting (Stream): a ASC, b ASC
+Sorting: __table1.a ASC
```

Comment:
I think this is correct because for a query `select distinct b, a from distinct_in_order_explain order by a` you don't get `(a, b)` order after sorting, even if read-in-order with `(a, b)`.

This is because the sorting step does not know anything about `b` and will use a stable sort (e.g. will prefer the left most input vs input with smallest `b`) which will break order.